src/System.Private.DataContractSerialization/src/System/Xml/XmlBaseWriter.cs

```diff
@@ -66,6 +66,14 @@ public override void Flush()
             _writer.Flush();
         }
 
+        public override Task FlushAsync()
+        {
+            if (IsClosed)
+                ThrowClosed();
```

Comment:
> Does this mean I should add this check in every public sync/async method?

@krwq, what would you recommend here?  I'm not familiar with the expectations around the design for the checking of in-flight operations.